envoy/server/factory_context.h

```diff
@@ -179,12 +179,35 @@ class ServerFactoryContext : public virtual CommonFactoryContext {
   virtual envoy::config::bootstrap::v3::Bootstrap& bootstrap() PURE;
 };
 
+/**
+ * Factory context for access loggers that need access to listener properties.
+ */
+class AccessLogFactoryContext : public virtual CommonFactoryContext {
```

Comment:
Up to you but I might call this `ListenerAccessLogFactoryContext` since I guess the idea is this is only usable when in listener context? For example it might not work for upstream access logs?

Also, I'm a little confused on how this is going to be used? I assume you are invoking this somehow from a private filter somewhere? Can you either add more notes so someone doesn't delete this later because it doesn't seem used or add a small integration test that shows how it could be used with a test filter? Up to you.

Reply:
Done